### PR TITLE
Dont build defintion with Gemfile.lock (if present) when resolving gems via inline gemfile

### LINKED ISSUE #5117: bundler/inline fails to install missing gems if Gemfile exists in working dir
If I am in directory that contains Gemfile and Gemfile.lock, running a bundler/inline script does not work as I expect.
### Expected behavior

Any missing gems in the inline gem declaration will be installed.
### Actual behavior

`Bundler::GemNotFound` is raised.
### Tested with

Bundler 1.13.6
Ruby 2.3.1
RubyGems 2.6.7

---

```
$ cat Gemfile
source "https://rubygems.org"
gem "rake"

$ cat test.rb
#!/usr/bin/env ruby
require "bundler/inline"
gemfile do
  source "https://rubygems.org"
  gem "whirly"
end
puts "Done"

$ bundle install
$ ruby test.rb
/Users/mbrictson/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/resolver.rb:366:in `block in verify_gemfile_dependencies_are_found!': Could not find gem 'whirly' in any of the gem sources listed in your Gemfile or available on this machine. (Bundler::GemNotFound)
    from /Users/mbrictson/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/resolver.rb:341:in `each'
    from /Users/mbrictson/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/resolver.rb:341:in `verify_gemfile_dependencies_are_found!'
    from /Users/mbrictson/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/resolver.rb:199:in `start'
    from /Users/mbrictson/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/resolver.rb:181:in `resolve'
    from /Users/mbrictson/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/definition.rb:250:in `resolve'
    from /Users/mbrictson/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/definition.rb:174:in `specs'
    from /Users/mbrictson/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/installer.rb:171:in `ensure_specs_are_compatible!'
    from /Users/mbrictson/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/installer.rb:79:in `run'
    from /Users/mbrictson/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/installer.rb:24:in `install'
    from /Users/mbrictson/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.13.6/lib/bundler/inline.rb:63:in `gemfile'
    from test.rb:3:in `<main>'
```

Removing the lock file from the working directory fixes the problem:

```
$ rm Gemfile.lock
$ ruby test.rb
Done
```

### PR DESCRIPTION
**Pull request by colby-swandale**
This fixes #5117 

I am passing an option to `Installer` to not build a definition when resolving gems via inline gemfile.

Also, i notice: `Bundler.default_lockfile.exist? && !options["update"]...`. Is "update" suppose to be a string? I ask because i noticed that other options are being passed as symbols, so I'm not which i'm suppose to be using.

Thanks.

PS. Hopefully i have the bundler terminology correct 😄

### CODE REVIEW

File: spec/runtime/inline_spec.rb [hunk 1]
```diff
@@ -193,4 +193,38 @@ def confirm(msg, newline = nil)
...
     expect(exitstatus).to be_zero if exitstatus
   end
+
+  it "installs inline gems when a Gemfile.lock is present" do
+    gemfile <<-G
+      source "https://rubygems.org"
+      gem "rake"
+    G
+
+    gemfile <<-G
```
Comment (segiddins) [inline spec/runtime/inline_spec.rb]: This should be `lockfile`?

Comment (colby-swandale) [inline spec/runtime/inline_spec.rb]: yep, thanks for catching that.

File: spec/runtime/inline_spec.rb [hunk 2]
```diff
@@ -193,4 +193,38 @@ def confirm(msg, newline = nil)
...
+      DEPENDENCIES
+        rake
+
+      BUNDLED WITH
+         1.13.6
+    G
+
+    script <<-RUBY
+      gemfile do
+        source "https://rubygems.org"
```
Comment (segiddins) [inline spec/runtime/inline_spec.rb]: this can't pull from an internet source, it should use one of the sources we build in the specs

Comment (colby-swandale) [inline spec/runtime/inline_spec.rb]: @segiddins are you referring to using the `gem_repo1` as the source?
```
gemfile do
  source "file://#{gem_repo1}"
  gem "rack"
end
```
?

Comment (segiddins) [inline spec/runtime/inline_spec.rb]: Yes

**Comment by segiddins** [comment]
@homu r+

**Comment by segiddins** [comment]
@bundlerbot r+

**Pull request merged**